### PR TITLE
issue 1060 port decryptFile() to Kotlin

### LINKED ISSUE #1060: implement pgp decrypt method that mimics error behavior
part of #1051 , needs to be done before doing #1057

### PR DESCRIPTION
**Pull request by IvanPizhenko**
Draft PR for decryption

close #1060

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)
- Difficult to test (explain why)
- Not worth testing
- Tests will be added later (issue #...)
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities

### CODE REVIEW

File: FlowCrypt/src/test/java/com/flowcrypt/email/security/pgp/PgpArmorTest.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+5BnI
+=F2om
+-----END PGP SIGNATURE----- 
+"""
+
+  private val messageText =
+      "Standard message\n\nsigned inline\n\nshould easily verify\nThis is email footer\n"
+
+  @Test
+  fun testReadClearTextSignedMessage() {
```
Comment (IvanPizhenko) [inline FlowCrypt/src/test/java/com/flowcrypt/email/security/pgp/PgpArmorTest.kt:48]: this test works

File: FlowCrypt/src/test/java/com/flowcrypt/email/security/pgp/PgpMsgTest.kt [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+
+  @Test
+  fun decryptionTest15() {
+    Assert.assertThrows(PgpMsg.NoMdcError::class.java) {
+      processMessage(messages["decrypt - encrypted missing checksum"]!!)
+    }
+  }
+
+  @Test
+  fun decryptionTest1() {
```
Comment (IvanPizhenko) [inline FlowCrypt/src/test/java/com/flowcrypt/email/security/pgp/PgpMsgTest.kt:515]: this test decrypts incorrectly

**Review by tomholub — COMMENTED** [review]
This code is indeed too low level for our purposes here - the libraries are too different to try to match code this closely. I was suspecting you made a harder task for yourself then necessary :)

You could probably achieve the desired purpose (80% of it) with a lot shorter function, that uses standard method to decrypt with PGPainless. Basically:

```kt
fun decrypt(datastream, privateKeys[]) {
  if(isArmored(dataStream)) {
    ... unarmor
  }
  try {
    r = pgpainlessdecrypt(stream, keys)
    return DecryptResult(...)
  } catch(e) {
    if(e ... ) {
      return DecryptionResult(error=MessageFormatError(...))
    } else if (e ...) {
      return DecryptionResult(error=MissingPassPhraseError(...))
    } else if (...) {
      ...
    } else {
      return DecryptResult( some unknown error - or maybe just throw? )
    }
  }
}
```

Maybe you cannot catch all of these situations. Then we have to find a way to improve pgpainless itself, so that you can.

For now, signature verification and the LongIds are out of scope for easier implementation.

**Comment by tomholub** [comment]
It doesn't have to work with streams yet actually, for further simplification.

**Pull request closed without merging**